packages/cli/generation/ir-generator/src/filtered-ir/IrGraph.ts

```diff
@@ -25,18 +26,21 @@ import {
     TypeNode,
 } from "./ids";
 
-export class AudienceIrGraph {
+export class IrGraph {
     private types: Record<TypeId, TypeNode> = {};
-    private typesNeededForAudience: Set<TypeId> = new Set();
     private errors: Record<TypeId, ErrorNode> = {};
     private endpoints: Record<EndpointId, EndpointNode> = {};
+    private audiences: Set<AudienceId> = new Set();
+    private servicesReferencedByType: Record<TypeId, Set<ServiceId>> = {};
```

Comment:
i think naming this `typesUsedByService` might make the code more readable

Reply:
Sweet, I went with `typesReferencedByService` so that its consistent with the other suggestion.